lib/SIL/IR/SILDeclRef.cpp

```diff
@@ -487,7 +487,8 @@ IsSerialized_t SILDeclRef::isSerialized() const {
 
   // Stored property initializers are inlinable if the type is explicitly
   // marked as @frozen.
-  if (isStoredPropertyInitializer() || isPropertyWrapperBackingInitializer()) {
+  if (isStoredPropertyInitializer() || (isPropertyWrapperBackingInitializer() &&
+      !d->getDeclContext()->isLocalContext())) {
```

Comment:
Since the below code traps if `cast<NominalTypeDecl>` fails, it's arguably clearer to call !isTypeContext() instead of isLocalContext(), or to replace the cast with a dyn_cast.